packages/js/package.json

```
@@ -3,12 +3,19 @@
   "version": "9.12.0",
   "description": "ESLint JavaScript language implementation",
   "main": "./src/index.js",
-  "scripts": {},
+  "types": "./types/index.d.ts",
+  "scripts": {
+    "test:types": "tsc -p tests/types/tsconfig.json"
+  },
   "files": [
     "LICENSE",
     "README.md",
-    "src"
+    "src",
+    "types"
   ],
+  "dependencies": {
+    "eslint": "^9.12.0"
+  },
```

Is it necessary to declare `eslint` as a dependency? For example, `@eslint/markdown` uses ESLint types but doesn't have `eslint` in `dependencies`.

---

What if someone uses the latest `@eslint/js` version but an older version of eslint that doesn't have the required types? If not a dependency, at least we should have it as a peer dependency (maybe we can mark it optional too).

---

I'd rather avoid introducing a circular dependency (`@eslint/js` is a dependency of `eslint`) especially since it's only needed for types, but I'm not sure what would the correct approach here. @eslint/eslint-team thoughts?

---

I'm not sure about the correct approach here. Open to suggestions. Is there any case where @eslint/js can be used independently without eslint?

---

I believe `eslint` should be an optional peer dependency of `@eslint/js` since `@eslint/js` is meant to be used with `eslint` but should not provide its own version of `eslint`. So in this case we should have:

```suggestion
  "devDependencies": {
    "eslint": "^9.13.0"
  },
  "peerDependencies": {
    "eslint": "^9.13.0"
  },
  "peerDependenciesMeta": {
    "eslint": {
      "optional": true
    }
  },
```

---

i think it's fine to not have `eslint` specified as a dependency.